js/util/script_detection.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const ideographicBreakingRegExp = new RegExp([
```

Comment:
If #3438 lands first, `verticalRegExp` and `ideographicBreakingRegExp` will have some overlap. To wit:

* Hangul and Mongolian are vertical but don’t get ideographic breaking.
* Halfwidth forms and Yi are horizontal but do get ideographic breaking.